File: javascript/bob/bob.js
module.exports = function Bob() {

  function isYelling(phrase) {
    return /[a-z]/i.test(phrase) && phrase.toUpperCase() === phrase;
  }

  function isSilence(phrase) {
    return !phrase.trim().length;
  }

  function isQuestion(phrase) {
    return /\?$/.test(phrase);
  }

  function hey(phrase) {
    if (isYelling(phrase)) 
      return 'Whoa, chill out!';

    if (isSilence(phrase)) 
      return 'Fine. Be that way!';
    
    if (isQuestion(phrase)) 
      return 'Sure.';

    return 'Whatever.'
  }

  return {
    hey: hey
  };
};

File: clojure/bob/src/bob.clj
(ns bob
  (use [clojure.string])
)

(defn any-char? [text]
  (boolean (re-find #"[a-zA-Z]+" text))
)

(defn uppercase? [text]
  (= text (upper-case text))
)

(defn yelling? [text]
  (if
    (any-char? text)
    (uppercase? text)
    false
  )
)

(defn question? [text]
  (ends-with? text "?")
)

(defn response-for [req]
  (cond 
    (blank? req) "Fine. Be that way!"
    (yelling? req) "Whoa, chill out!"
    (question? req) "Sure."
    :else "Whatever."
  )
)